lib/src/main/java/com/stepango/rxdatabindings/DatabindingsExt.kt

```diff
@@ -26,57 +26,31 @@ import android.databinding.ObservableLong
 import android.databinding.ObservableParcelable
 import android.databinding.ObservableShort
 import android.os.Parcelable
+import android.text.Spanned
 import io.reactivex.Observable
 import io.reactivex.Scheduler
 import android.databinding.Observable as DataBindingObservable
 
 
-internal inline fun <T : DataBindingObservable, R : Any?> T.observe(
-    scheduler: Scheduler?,
-    fireInitialValue: Boolean,
-    crossinline transformer: (T) -> R
-): Observable<R> = Observable.create<R> { source ->
-
-    if (fireInitialValue && !source.isDisposed) try {
-        source.onNext(transformer(this))
-    } catch (e: Exception) {
-        source.onError(e)
-        return@create
-    }
-
-    object : DataBindingObservable.OnPropertyChangedCallback() {
-        @Suppress("UNCHECKED_CAST")
-        override fun onPropertyChanged(observable: DataBindingObservable, id: Int) = if (!source.isDisposed) try {
-            source.onNext(transformer(observable as T))
-        } catch (e: Exception) {
-            source.onError(e)
-        } else Unit
-    }.let {
-        source.setCancellable { removeOnPropertyChangedCallback(it) }
-        addOnPropertyChangedCallback(it)
-    }
-
-}.let { if (scheduler != null) it.subscribeOn(scheduler) else it }
-
-internal inline fun <T : DataBindingObservable, R : Any> T.safeObserve(
+internal inline fun <T : DataBindingObservable, R : Any> T.observe(
     scheduler: Scheduler?,
     fireInitialValue: Boolean,
     crossinline transformer: (T) -> R?
 ): Observable<R> = Observable.create<R> { source ->
 
     if (fireInitialValue && !source.isDisposed) try {
-        transformer(this)?.let { source.onNext(it) }
-    } catch (e: Exception) {
-        source.onError(e)
+        transformer(this)?.let(source::onNext)
+    } catch (throwable: Throwable) {
+        source.onError(throwable)
         return@create
     }
 
     object : DataBindingObservable.OnPropertyChangedCallback() {
         @Suppress("UNCHECKED_CAST")
         override fun onPropertyChanged(observable: DataBindingObservable, id: Int) = if (!source.isDisposed) try {
-            transformer(observable as T)?.let { source.onNext(it) } ?: Unit
-        } catch (e: Exception) {
-            source.onError(e)
+            transformer(observable as T)?.let(source::onNext) ?: Unit
```

Comment:
Should we throw an error here?

Reply:
I was thinking about the same, but I believe it would be a more common use case to not throw error here. May be we can provide a param to change the default behaviour.